File: javascript/scripts/export-contract-results.mjs
import { readFileSync, writeFileSync } from "node:fs";
import { resolve } from "node:path";
import { fileURLToPath } from "node:url";
import { evaluateExpression } from "../dist/expr.js";
import { loadConfigFromDict } from "../dist/config.js";

const root = resolve(fileURLToPath(new URL(".", import.meta.url)), "../../");
const expressions = {};
for (const c of JSON.parse(readFileSync(resolve(root, "tests/parity/expression_cases.json")))
  .expression_cases) {
  try {
    expressions[c.name] = evaluateExpression(c.expr, c.vars ?? {}).toFixed(4);
  } catch {
    expressions[c.name] = "error";
  }
}
const configs = {};
for (const c of JSON.parse(readFileSync(resolve(root, "tests/parity/config_validation_cases.json")))
  .cases) {
  try {
    loadConfigFromDict(c.config);
    configs[c.name] = "accept";
  } catch {
    configs[c.name] = "reject";
  }
}
writeFileSync(process.argv[2], `${JSON.stringify({ expressions, configs }, null, 2)}\n`);

File: python/scripts/export_contract_results.py
"""Emit deterministic SDK contract results for the cross-language CI gate."""

from __future__ import annotations

import json
import sys
from decimal import ROUND_HALF_UP, Decimal
from pathlib import Path

from bursar.config import load_config_from_dict
from bursar.expr import ExpressionError, evaluate_expression

ROOT = Path(__file__).parents[2]


def main(output: str) -> None:
    expression_cases = json.loads((ROOT / "tests/parity/expression_cases.json").read_text())["expression_cases"]
    config_cases = json.loads((ROOT / "tests/parity/config_validation_cases.json").read_text())["cases"]
    expressions: dict[str, str] = {}
    for case in expression_cases:
        try:
            value = evaluate_expression(case["expr"], case.get("vars", {}))
            expressions[case["name"]] = f"{value.quantize(Decimal('0.0001'), rounding=ROUND_HALF_UP):.4f}"
        except ExpressionError:
            expressions[case["name"]] = "error"

    configs: dict[str, str] = {}
    for case in config_cases:
        try:
            load_config_from_dict(case["config"])
            configs[case["name"]] = "accept"
        except Exception:
            configs[case["name"]] = "reject"

    Path(output).write_text(json.dumps({"expressions": expressions, "configs": configs}, sort_keys=True) + "\n")


if __name__ == "__main__":
    main(sys.argv[1])
